src/formatDate.js

```diff
@@ -8,7 +8,60 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3]; // For example "/"
+  const newSeparator = toFormat[3]; // For example "."
+  const newFormat = toFormat;
+  const newFormatSorted = [...toFormat].sort(); // for easy found YYYY or YY ??
+  let YearFormFromFormat = ''; // found what we in old format YY OR YYYY ??
+  const arrayFromDataString = date.split(oldSeparator);
+  const objectOfDataAndFormats = {};
+  // object =  keys = index of fromFormat, values = digits of date;
+  const fromFormatLenghtWitoutSeparator = fromFormat.length - 1;
+  // delete Separator in the end of array
+  const arrayOfResult = [];
+
+  // many variables for easy reading, not )
+
+  if (fromFormat.includes('YY')) {
+    YearFormFromFormat = 'YY';
+  }
+
+  if (fromFormat.includes('YYYY')) {
+    YearFormFromFormat = 'YYYY';
+  }
+
+  for (let i = 0; i < fromFormatLenghtWitoutSeparator; i++) {
+    if (fromFormat[i] === YearFormFromFormat) {
+      fromFormat[i] = newFormatSorted[3];
+    }
+    objectOfDataAndFormats[fromFormat[i]] = arrayFromDataString[i];
+  }
+
+  let valueOfKey = objectOfDataAndFormats[newFormatSorted[3]];
+
+  if (valueOfKey.length > newFormatSorted[3].length) {
+    valueOfKey = valueOfKey.slice(valueOfKey.length - 2);
+
+    objectOfDataAndFormats[newFormatSorted[3]] = valueOfKey;
+  }
+
+  if (valueOfKey.length < newFormatSorted[3].length) {
+    if (valueOfKey < 30) {
+      valueOfKey = `20${valueOfKey}`;
+    } else {
+      valueOfKey = `19${valueOfKey}`;
+    }
+
+    objectOfDataAndFormats[newFormatSorted[3]] = valueOfKey;
+  }
+
+  for (const ch of newFormat) {
+    if (objectOfDataAndFormats[ch]) {
+      arrayOfResult.push(objectOfDataAndFormats[ch]);
+    }
+  }
+
+  return arrayOfResult.join(newSeparator);
```

Comment:
The function implementation seems to have a logical flaw in handling the year conversion. It is not considering the case when the year is in the middle of the format. Additionally, the sorting of the `newFormat` array could lead to incorrect formatting if the `toFormat` array is not in the order of 'YY', 'MM', 'DD'. The code also assumes that the year will always be the last element in the sorted array, which may not be the case. Consider reworking the logic to map the date components directly from `fromFormat` to `toFormat` without sorting, and ensure proper year conversion based on the position in the format array.